File: src/modules/common/components/decorators/minLength.ts
import decoratorHelper from "../../helpers/decoratorHelper"

export function minLength(messageKey: string, min: number) {
    return function (target: any, propertyKey: string) {
        return decoratorHelper.define(target, propertyKey, messageKey, isValid);
    }

    function isValid(value: any) {
        return !!value && value.length >= min;
    }
}

File: Rest/ExamERP/ExamERP.Common/Attributes/MinLength.cs
namespace ExamERP.Common.Attributes
{
    public class MinLength : BaseAttribute
    {
        public int Min { get; set; }
        public MinLength(string messageKey, int minLength) : base(messageKey)
        {
            this.Min = minLength;
        }
        public override bool IsValid(object value)
        {
            return value.ToString().Length >= this.Min;
        }
    }
}
